### PR TITLE
New feature: add string format "date time"

### PR DESCRIPTION
**Pull request by gigaga**

### CODE REVIEW

**Comment by codekie** [comment]
This feature has been extracted from #156

File: src/validator.js [hunk 1]
```diff
@@ -114,11 +117,16 @@ function _createReferenceSchema(specSchema) {
...
 function _addFormatValidators(validator) {
+    const dateTimeRegex = new RegExp(REGEX_DATE_TIME);
+
     validator.addFormat('int32', { type: 'number', validate: FormatValidator.int32 });
     validator.addFormat('int64', { type: 'string', validate: FormatValidator.int64 });
     validator.addFormat('float', { type: 'number', validate: FormatValidator.float });
     validator.addFormat('double', { type: 'number', validate: FormatValidator.double });
     validator.addFormat('byte', { type: 'string', validate: FormatValidator.byte });
+    validator.addFormat('date-time', {
+        validate: (dateTimeString) => dateTimeRegex.test(dateTimeString)
```
Comment (codekie) [inline src/validator.js:128]: I suggest to export the validator into a separate module (`src/validators/date-time.js`) which then exports a `validateDateTime`-function.

That way, more thorough unit-tests (`test/specs/validators/date-time.js`), additionally to the supplied integration tests, can be supplied.

Here is a list of of test-cases I have come up, so far:

**Valid**

```
-1001-08-30T01:45:36.123Z
2008-08-30T01:45:36.123Z
1985-04-12T23:20:50.52Z
1996-12-19T16:39:57-08:00
1937-01-01T12:00:27.87+00:20
2017-07-21T17:32:28Z
2007-08-31T16:47:00+00:00
2007-12-24T18:21:00Z
2009-01-01T12:00:00+01:00
2009-06-30T18:30:00+02:00
2021-12-01T13:54:15+00:00
2021-12-01T13:54:15Z
2021-12-01t13:54:15z
2024-03-30T12:34:56Z
```

**Leap-second**

```
1990-12-31T23:59:60Z
1990-12-31T15:59:60-08:00
```

**Leap year**

```
2024-02-29T12:34:56Z
```

**Invalid**

```
2008-08-30T01:45:36
108-08-30T01:45:36
2008-08-32T01:45:36
2008-08-02T27:45:36
2009-01-01T12:00:00+01:00:15
2024-02-30T12:34:56Z
2021-13-02T08:27:49Z
2022-07-32T09:36:24Z
2023-02-29T12:34:56Z
2021-12-01TZ
2021-12-01Z
2009-01-01T+01:00
1996-12-19-08:00
2009-06-30+02:00
```

**Comment by codekie** [comment]
@gigaga If you want, I can merge your PR into a separate feature-branch, in this repository, and I'll add my suggested changes on top of it.

**Comment by gigaga** [comment]
Hi @codekie 

> @gigaga If you want, I can merge your PR into a separate feature-branch, in this repository, and I'll add my suggested changes on top of it.

Yes, no problem for me :)

**Comment by codekie** [comment]
The validator has been integrated to the default-branch now. Thank you for your contribution 👍

**Comment by gigaga** [comment]
> The validator has been integrated to the default-branch now. Thank you for your contribution 👍

Wonderful... I am in an hurry to get you next release with discrimator feature ;)

**Pull request merged**